### PR TITLE
Modified scanner to consider more cases

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Scanner will consider unfinished tags and look for the next one
It also considers Prolog and ProcessingInstructions

***I also commented out the failing Diagnostics tests

### CODE REVIEW

File: extensions/org.eclipse.lsp4xml.contentmodel/src/test/java/org/eclipse/lsp4xml/contentmodel/xsd/XMLBasicDiagnosticsTest.java [hunk 1]
```diff
@@ -89,28 +89,28 @@ public void testDashDashInComment() throws Exception {
...
-		XMLLanguageService languageService = new XMLLanguageService();
-		String xml = new StringBuilder("<Id att=\"\"value\" att2=\"value2\">\r\n" + //
-				"        </Id>") //
-						.toString();
-		TextDocument document = new TextDocument(xml.toString(), "test.xml");
-		List<Diagnostic> diagnostics = languageService.doDiagnostics(document, () -> {
-		});
-		assertDiagnostics(diagnostics, d(1, 8, 1, 10, "emptyprefixedattname"));
-	}
+	// @Test
```
Comment (angelozerr) [inline extensions/org.eclipse.lsp4xml.contentmodel/src/test/java/org/eclipse/lsp4xml/contentmodel/xsd/XMLBasicDiagnosticsTest.java]: Please don't update XMLBasicDiganosticsTest. I'm working on it. (You can use @Ignore annotation to ignore test)

Comment (NikolasKomonen) [reply extensions/org.eclipse.lsp4xml.contentmodel/src/test/java/org/eclipse/lsp4xml/contentmodel/xsd/XMLBasicDiagnosticsTest.java]: Is it okay if I update the PR with @ignore to make sure the Travis tests still pass? Or do you want XMLBasicDiagnosticTests not touched at all?

Comment (angelozerr) [reply extensions/org.eclipse.lsp4xml.contentmodel/src/test/java/org/eclipse/lsp4xml/contentmodel/xsd/XMLBasicDiagnosticsTest.java]: Please don't touch it. I will fix it. Thanks

**Comment by NikolasKomonen** [comment]
@angelozerr I fixed the PR, would it be fine to merge it now?

**Comment by angelozerr** [comment]
Thanks @NikolasKomonen !

**Pull request merged**